hasktorch/src/Torch/Typed/Parameter.hs

```diff
@@ -30,22 +30,22 @@ import           GHC.Generics
 
 import qualified Torch.NN                      as A
 import qualified Torch.Autograd                as A
-import qualified Torch.Tensor                  as A
+import qualified Torch.Tensor                  as D
 import qualified Torch.DType                   as D
 import qualified Torch.Device                  as D
 import           Torch.Typed.Aux
 import           Torch.Typed.Factories
 import           Torch.Typed.Functional
 import           Torch.Typed.Tensor
 
-newtype Parameter (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: [Nat]) = Parameter A.IndependentTensor
-  deriving (Show)
+newtype Parameter (device :: (D.DeviceType, Nat)) (dtype :: D.DType) (shape :: [Nat]) = UnsafeMkParameter A.IndependentTensor
```

Comment:
Just wondering - what's the reasoning for marking this constructor as unsafe?

Reply:
Thanks for the question, it’s roughly the same as for the typed `Tensor`: with this constructor you can lift any `IndependentTensor` to a typed Parameter with arbitrary `device`, `dtype`, and `shape`. thus, the runtime values of these and the static values do not have agree which can lead to exceptions and undefined behaviour.